### PR TITLE
feat(ComboBox): L3-6763 ComboBox updates for user details page integration

### PR DESCRIPTION
**Pull request by smarks8**
**Jira ticket**

[L3-6763](https://phillipsauctions.atlassian.net/browse/L3-6763)

**Screenshots**
image

**Figma link**

n/a

**Summary**

Refactor ComboBox to support additional use cases: 
- Update to allow for custom labels 
- Ability to filter on additional terms 
- Adds invalid text for error handling 
- Updates buttons to use IconButton 

**Change List (describe the changes made to the files)**
This pull request introduces several enhancements and updates to the `ComboBox` component, including new features, improved examples, and expanded test coverage. The changes focus on improving usability, adding customization options, and ensuring robust functionality.

### Enhancements to `ComboBox` Examples
* Added new examples in `ComboBox.stories.tsx` to demonstrate advanced use cases, including custom label rendering, filtering with additional terms, and handling duplicate display values. These examples include `GetOptionLabelExample`, `CustomRendering`, `FilterTermsExample`, and `Phone`.
* Updated `Playground` to include a `value` state and expanded `argTypes` for better Storybook controls.
* Modified the `ComboBoxInDrawer` example to include `value` and `onChange` props for both inside and outside dropdowns. [[1]](diffhunk://#diff-20109226114674d39e1a4f4114f7f1432958a60632fb813452d922a8695557b1R74-R75) [[2]](diffhunk://#diff-20109226114674d39e1a4f4114f7f1432958a60632fb813452d922a8695557b1R90-R99)

### Expanded Test Coverage
* Replaced outdated tests in `ComboBox.test.tsx` with comprehensive test cases to cover new functionalities, such as filtering with `filterTerms`, custom option rendering, controlled mode behavior, and handling invalid states.
* Added tests for edge cases, including no matching options, dropdown toggling, and exact match selection on blur or Enter key.

### Codebase Improvements
* Refactored `ComboBox` examples to use the `ComboBoxOption` type for consistency and better type safety.
* Updated the `birthdays` example to generate years starting from 1990 instead of 1926 for a more relevant range.

These changes collectively enhance the `ComboBox` component's functionality, usability, and maintainability.

**Acceptance Test (how to verify the PR)**

- See that all storybook cases work as expected

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-6763]: https://phillipsauctions.atlassian.net/browse/L3-6763?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/ComboBox/ComboBox.test.tsx [hunk 1]
```diff
@@ -1,135 +1,314 @@
...
-    expect(screen.queryByText('No Options.')).toBeInTheDocument();
+    // Press down arrow to open dropdown
+    await userEvent.type(input, '{arrowdown}');
+
+    await waitFor(() => {
+      expect(screen.getByText('Apple')).toBeInTheDocument();
+    });
+
+    // Press escape to close dropdown
+    await userEvent.type(input, '{escape}');
```
[the remark this answers is not in the document]
Comment (DaveMcNamara) [reply src/components/ComboBox/ComboBox.test.tsx:155]: `userEvent.type` _should_ work for all interactions, though you might need `{ skipClick: true }` to stop it from clicking on the thing first. That said, I'm not really bothered by the mixing and matching of `type` and `keyboard` so long as `fireEvent` is nowhere to be seen.

File: src/components/PhoneNumberPicker/index.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default as PhoneNumberPicker, type PhoneNumberPickerProps } from './PhoneNumberPicker';
```
[the remark this answers is not in the document]
Comment (DaveMcNamara) [reply src/components/PhoneNumberPicker/index.ts:1]: `CountryCodeAutocomplete`?

File: src/components/PhoneNumberPicker/PhoneNumberPicker.test.tsx [hunk 3]
```diff
@@ -1,57 +1,104 @@
...
+    render(<PhoneNumberPickerWrapper />);
 
     const input = screen.getByTestId('test-id-combobox-input');
+
+    // Type directly in the input
     await userEvent.type(input, '+1');
-    await userEvent.keyboard('{Enter}');
 
+    // Find the US option
+    const usOption = screen.getByText(/\(US\)/i);
```
[the remark this answers is not in the document]
Comment (DaveMcNamara) [reply src/components/PhoneNumberPicker/PhoneNumberPicker.test.tsx:80]: Oh wait, I'm wrong. I read those parentheses as a RegEx capture group, but they are escaped. Never mind on this comment.

File: src/components/ComboBox/ComboBox.tsx [hunk 4]
```diff
@@ -36,13 +35,46 @@ export interface ComboBoxProps {
...
+  value?: string;
+  /**
+   * Handler for value changes
+   */
+  onChange?: (value: string, option: ComboBoxOption | null) => void;
+  /**
+   * Input value (display text)
+   * Only needed for controlled usage, otherwise managed internally
+   */
+  inputValue?: string;
```
Comment (scottdickerson) [inline src/components/ComboBox/ComboBox.tsx:53]: I don't understand why we need a `inputValue` and `setInputValue` in addition to the `onChange` and `value` prop?

File: src/components/ComboBox/ComboBox.tsx [hunk 5]
```diff
@@ -36,13 +35,46 @@ export interface ComboBoxProps {
...
+  /**
+   * Function to get display text for an option
+   * Similar to MUI's getOptionLabel
+   */
+  getOptionLabel?: (option: ComboBoxOption) => string;
+  /**
+   * Function to render custom option content
+   * Similar to MUI's renderOption
+   */
+  renderOption?: (option: ComboBoxOption) => React.ReactNode;
```
Comment (scottdickerson) [inline src/components/ComboBox/ComboBox.tsx:68]: why not just let the option "displayValue" be a ReactNode instead?

File: src/components/ComboBox/types.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+export interface ComboBoxOption {
+  /**
+   * The label or display text
+   */
+  label?: string;
+  /**
+   * The option value (used for selection)
+   */
+  value: string;
+  /**
```
Comment (scottdickerson) [inline src/components/ComboBox/types.ts:10]: probably should add a comment here explaining that this is for usecases where the display value inside the input after selecting an option doesn't exactly match what the option looked like in the dropdown.

File: src/components/ComboBox/types.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+export interface ComboBoxOption {
+  /**
+   * The label or display text
+   */
+  label?: string;
```
Comment (scottdickerson) [inline src/components/ComboBox/types.ts:5]: could we just make this and `displayValue` a `ReactNode`?

File: src/components/ComboBox/ComboBox.tsx [hunk 8]
```diff
@@ -1,11 +1,13 @@
 import * as Popover from '@radix-ui/react-popover';
 import classnames from 'classnames';
 import { Command, CommandGroup, CommandInput, CommandItem, CommandList } from 'cmdk';
-import React from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { useOnClickOutside } from 'usehooks-ts';
 import * as iconComponents from '../../assets/formatted';
```
Comment (scottdickerson) [inline src/components/ComboBox/ComboBox.tsx:6]: this should be an explicit import for each icon otherwise code-splitting and tree-shaking doesn't work correctly

**Review by scottdickerson — APPROVED** [review]
LGTM

**Comment by constantinehuzenko** [comment]
is this hover effect consistent? should input be hovered when button hovered? It's not a blocker just question
for reference https://mui.com/material-ui/react-autocomplete/

https://github.com/user-attachments/assets/abefbda9-b692-4a0e-89bd-410ca9da4ac0

**Comment by constantinehuzenko** [comment]
tab does't go to cancel icon/button
https://github.com/user-attachments/assets/35a781d6-60f7-4869-939e-4a79362b071a

**Review by constantinehuzenko — APPROVED** [review]
